src/serve_rendered.js

```diff
@@ -1471,13 +1521,24 @@ export const serve_rendered = {
         const j = Math.min(maxPoolSizes.length - 1, s - 1);
         const minPoolSize = minPoolSizes[i];
         const maxPoolSize = Math.max(minPoolSize, maxPoolSizes[j]);
-        map.renderers[s] = createPool(s, 'tile', minPoolSize, maxPoolSize);
-        map.renderers_static[s] = createPool(
+        const languages = new Set([null, ...(options.languages || [])]);
+
+        map.renderers[s] = createPool(
           s,
-          'static',
+          'tile',
           minPoolSize,
           maxPoolSize,
+          null,
         );
+        for (const language of languages) {
+          map.renderers_static[`${s}${language}`] = createPool(
+            s,
+            'static',
+            language ? 1 : minPoolSize,
+            language ? 1 : maxPoolSize,
+            language,
+          );
+        }
```

Comment:
I'd caution again adding to many more pools than the user specifies. I think I already pushed that a bit doubling the pool size for tile/static mode, 

Sometimes that maxPoolSizes is important for lower power systems, which need to limit the number of cores that can be used.

Reply:
Yeah, pool handling was the main point where I had doubts. I suspected the lower power constraint, but needed confirmation and details, so thanks.
I haven't tried calling `renderer.load()` before each render. That's what you are suggesting, right? I was afraid of the performance penalty. That's the only reason we have separate pools for scale ratios and for 'tile' vs 'static', right? Instead of one big pool with dynamic configuration before each render.
If I remember correctly, `renderer.load()` causes requests of some assets.
What should I investigate first in your opinion?

Reply:
The separate pools for static vs tile mode have two do with two issues. 

when used as a tile server, maplibre-native tile mode is important to prevent label clipping. when in tile mode, maplibre considers information beyond the edges of the tile, which helps with labels that may go beyond the edge (note that the source has to support this bit of extra data, like plantiler and openmaptiles tools do , but tilemaker does not)

However we have found that recent changes in maplibre break static images when in tile mode. the reason is that a change has been made when it was still mapbox-native that limit tile mode to only return a single tile. since a static images can span multiples tiles, the resulting image ends up with missing data. to work around this I made static images render in maplibre 'static' mode, which can span multiple tiles. (but does not deal with label clipping in any way)

Reply:
Yes, I don't doubt we need to have different static and tile modes.
But why can't we have a single pool that constructs a new `mlgl.Map` for each render? Is that really slow?
I see now that the situation is different for the style (`.load()` method) than it is for the mode and the scale (which have to be set in the `Map` constructor). But if it's slow to build new `mlgl.Map` dynamically, is calling `.load()` slow too?

Reply:
The pool is needed because you cannot render multiple images on the same map, that is, each map can render ony one image at a time. And you need different map instances for each resolution, because it's something you configure when instantiating the map. That said, there are some strategies for rendering different styles (or the same style with different language). Creating lots of instances (using several pools) can lead to consuming lots of memory. One solution is loading the style before the rendering pass, like @acalcutt suggested. This will lead to cache to be cleared everytime, so you will may have performance issues. But I think it's better than creating lots of map instances, unless you specify minPoolSize as 0, so you will create map instances only if you really use them.